dual_screen/android/src/main/kotlin/com/builttoroam/dual_screen/DualScreenPlugin.kt

```diff
@@ -0,0 +1,132 @@
+
+/** DualScreenPlugin */
+public class DualScreenPlugin :
+    FlutterPlugin,
+    MethodCallHandler,
+    EventChannel.StreamHandler,
+    ActivityAware {
+    private val METHOD_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/methods"
+    private val EVENT_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/events"
+    private lateinit var methodChannel: MethodChannel
+    private lateinit var eventChannel: EventChannel
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    fun registerPlugin(context: Context?, messenger: BinaryMessenger?) {
+        activity = this.activity
+        this.context = context
+        MethodChannel(messenger, METHOD_CHANNEL_NAME).setMethodCallHandler(this)
+        EventChannel(messenger, EVENT_CHANNEL_NAME).setStreamHandler(this)
+    }
+
+    override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        this.registerPlugin(binding.getApplicationContext(), binding.getBinaryMessenger())
+    }
+
+    companion object {
+        @JvmStatic
+        fun registerWith(registrar: Registrar) {
+            DualScreenPlugin().registerPlugin(registrar.context(), registrar.messenger())
+        }
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "isDualScreenDevice" -> {
+                result.success(isDualScreenDevice())
+            }
+            "isAppSpanned" -> {
+                if (isDualScreenDevice()) {
+                    result.success(isAppSpanned())
+                } else {
+                    result.success(false)
+                }
+            }
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+    override fun onListen(arguments: Any?, events: EventSink?) {
+        activity?.getWindow()?.getDecorView()?.getRootView()?.addOnLayoutChangeListener(object : OnLayoutChangeListener {
+            override fun onLayoutChange(
+                v: View?,
+                left: Int,
+                top: Int,
+                right: Int,
+                bottom: Int,
+                oldLeft: Int,
+                oldTop: Int,
+                oldRight: Int,
+                oldBottom: Int
+            ) {
+                events!!.success(isAppSpanned())
+            }
+        })
+    }
+
+    override fun onCancel(arguments: Any?) {
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        this.activity = null
+    }
+
+    override fun onDetachedFromActivity() {
+        this.activity = null
+    }
+
+    fun isDualScreenDevice(): Boolean {
+        return context?.getPackageManager()!!.hasSystemFeature(
```

Comment:
Isn't this going to crash the app if `getPackageManger` will be null?

Reply:
Yep you're right, I've added in a check